ciao-controller/main.go

```diff
@@ -283,3 +295,33 @@ func (c *controller) startCiaoService() error {
 
 	return nil
 }
+
+func getControllerFQDNFromCert(cert string) (string, error) {
+	bytesCert, err := ioutil.ReadFile(cert)
+	if err != nil {
+		return "", fmt.Errorf("couldn't read cert: %s", err)
+	}
+
+	blockCert, _ := pem.Decode(bytesCert)
+	if blockCert == nil {
+		return "", fmt.Errorf("couldn't decode cert")
+	}
+
+	parsedCert, err := x509.ParseCertificate(blockCert.Bytes)
+	if err != nil {
+		return "", fmt.Errorf("couldn't parse cert: %s", err)
+	}
+
+	role := ssntp.GetRoleFromOIDs(parsedCert.UnknownExtKeyUsage)
+	if !role.IsController() {
+		return "", fmt.Errorf("expected certificate role ssntp.Controller, got role %s", role.String())
```

Comment:
I'd looked at a sample https cert from the controller web and was seeing slightly different data in it. Initially it was the one I was going to go for.  But I liked that the ssntp one enabled the sanity check of the role.  Now that I think about it though, the ssntp one is possibly...or even probably most desirably...on a different network and with a different name than the visible end-point name for the user access API's.  I think now this MUST use the https api web cert.